rotors_gazebo_plugins/src/gazebo_motor_model.cpp

```diff
@@ -104,47 +117,140 @@ void GazeboMotorModel::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf) {
-  ROS_ASSERT_MSG(rot_velocities->angular_velocities.size() > motor_number_,
-                 "You tried to access index %d of the MotorSpeed message array which is of size %d.",
-                 motor_number_, rot_velocities->angular_velocities.size());
-  ref_motor_rot_vel_ = std::min(rot_velocities->angular_velocities[motor_number_], max_rot_velocity_);
+void GazeboMotorModel::CreatePubsAndSubs() {
+
+  gzdbg << __PRETTY_FUNCTION__ << " called." << std::endl;
+
+  // Create temporary "ConnectGazeboToRosTopic" publisher and message
+  gazebo::transport::PublisherPtr gz_connect_gazebo_to_ros_topic_pub =
+      node_handle_->Advertise<gz_std_msgs::ConnectGazeboToRosTopic>("~/" + kConnectGazeboToRosSubtopic, 1);
+  gz_std_msgs::ConnectGazeboToRosTopic connect_gazebo_to_ros_topic_msg;
+
+  // Create temporary "ConnectRosToGazeboTopic" publisher and message
+  gazebo::transport::PublisherPtr gz_connect_ros_to_gazebo_topic_pub =
+      node_handle_->Advertise<gz_std_msgs::ConnectRosToGazeboTopic>("~/" + kConnectRosToGazeboSubtopic, 1);
+  gz_std_msgs::ConnectRosToGazeboTopic connect_ros_to_gazebo_topic_msg;
+
+  // ============================================ //
+  //  ACTUAL MOTOR SPEED MSG SETUP (GAZEBO->ROS)  //
+  // ============================================ //
+
+  motor_velocity_pub_ = node_handle_->Advertise<gz_std_msgs::Float32>("~/" + namespace_ + "/" + motor_speed_pub_topic_, 1);
+
+  // Connect to ROS
+  //connect_gazebo_to_ros_topic_msg.set_gazebo_namespace(namespace_);
+  connect_gazebo_to_ros_topic_msg.set_gazebo_topic("~/" + namespace_ + "/" + motor_speed_pub_topic_);
+  connect_gazebo_to_ros_topic_msg.set_ros_topic(namespace_ + "/" + motor_speed_pub_topic_);
+  connect_gazebo_to_ros_topic_msg.set_msgtype(gz_std_msgs::ConnectGazeboToRosTopic::FLOAT_32);
+  gz_connect_gazebo_to_ros_topic_pub->Publish(connect_gazebo_to_ros_topic_msg, true);
+
+  // ============================================ //
+  // = CONTROL VELOCITY MSG SETUP (ROS->GAZEBO) = //
+  // ============================================ //
+
+  command_sub_ = node_handle_->Subscribe("~/" + namespace_ + "/" + command_sub_topic_, &GazeboMotorModel::ControlVelocityCallback, this);
+
+  connect_ros_to_gazebo_topic_msg.set_ros_topic(namespace_ + "/" + command_sub_topic_);
+  //connect_ros_to_gazebo_topic_msg.set_gazebo_namespace(namespace_);
+  connect_ros_to_gazebo_topic_msg.set_gazebo_topic("~/" + namespace_ + "/" + command_sub_topic_);
+  connect_ros_to_gazebo_topic_msg.set_msgtype(gz_std_msgs::ConnectRosToGazeboTopic::COMMAND_MOTOR_SPEED);
+  gz_connect_ros_to_gazebo_topic_pub->Publish(connect_ros_to_gazebo_topic_msg, true);
+
+  // ============================================ //
+  // ==== WIND SPEED MSG SETUP (ROS->GAZEBO) ==== //
+  // ============================================ //
+
+  wind_speed_sub_ = node_handle_->Subscribe("~/" + namespace_ + "/" + wind_speed_sub_topic_, &GazeboMotorModel::WindSpeedCallback, this);
```

Comment:
do we need windspeed here? I think we have a wind_speed_plugin that should connect to the ros topics.